### ISSUE TITLE
ONNXRuntimeError in optimum / m2m100_418M

### ISSUE THREAD

**Issue by goga334** [huggingface/optimum#1053]
### System Info

- `transformers` version: 4.29.1
- Platform: Linux-5.10.147+-x86_64-with-glibc2.31
- Python version: 3.10.11
- Huggingface_hub version: 0.14.1
- Safetensors version: not installed
- PyTorch version (GPU?): 2.0.0+cu118 (False)
- Tensorflow version (GPU?): 2.12.0 (False)
- Flax version (CPU?/GPU?/TPU?): 0.6.9 (cpu)
- Jax version: 0.4.8
- JaxLib version: 0.4.7
- Using GPU in script?: no
- Using distributed or parallel set-up in script?: no

### Who can help?

@sgugger

### Information

- [X] The official example scripts
- [X] My own modified scripts

### Tasks

- [X] An officially supported task in the `examples` folder (such as GLUE/SQuAD, ...)
- [ ] My own task or dataset (give details below)

### Reproduction

In google colab I try to use example from here: https://huggingface.co/optimum/m2m100_418M

```
from transformers import AutoConfig, AutoTokenizer
from optimum.onnxruntime import ORTModelForSeq2SeqLM`
from optimum.pipelines import pipeline

hi_text = "जीवन एक चॉकलेट बॉक्स की तरह है।"
chinese_text = "生活就像一盒巧克力。"

model = ORTModelForSeq2SeqLM.from_pretrained("optimum/m2m100_418M")
tokenizer = AutoTokenizer.from_pretrained("optimum/m2m100_418M")

# translate Hindi to French
tokenizer.src_lang = "hi"
encoded_hi = tokenizer(hi_text, return_tensors="pt")
generated_tokens = model.generate(**encoded_hi, forced_bos_token_id=tokenizer.get_lang_id("fr"))
```

And in the last line I get such error:

```
RuntimeException: [ONNXRuntimeError] : 6 : RUNTIME_EXCEPTION : Non-zero status code returned while running Add node. Name:'Add_749' Status Message: /onnxruntime_src/onnxruntime/core/providers/cpu/math/element_wise_ops.h:540 void onnxruntime::BroadcastIterator::Init(ptrdiff_t, ptrdiff_t) axis == 1 || axis == largest was false. Attempting to broadcast an axis by a dimension other than 1. 3 by 15
```

### Expected behavior

I expected smooth work and same results as in the tutorial.

**Comment by michaelbenayoun** [comment]
Hi can you try exporting the m2m100 model with the Optimum CLI?

```
optimum-cli export onnx -m facebook/m2m100_418M --task seq2seq-lm-with-past m2m100_onnx
```

Then use this exported model:

```python
model = ORTModelForSeq2SeqLM.from_pretrained("m2m100_onnx")
```

Do you still have the issue?

**Comment by goga334** [comment]
@michaelbenayoun, hi, thanks for help) I received following exception while exporting:
```
Exception: An error occured during validation, but the model was saved nonetheless at m2m100_onnx. Detailed error: [ONNXRuntimeError] : 1 : FAIL : Load model from m2m100_onnx/decoder_model_merged.onnx failed:/onnxruntime_src/onnxruntime/core/graph/model.cc:146 onnxruntime::Model::Model(onnx::ModelProto&&, const PathString&, const IOnnxRuntimeOpSchemaRegistryList*, const onnxruntime::logging::Logger&, const onnxruntime::ModelOptions&) Unsupported model IR version: 9, max supported IR version: 8
.
```
By the way, I've met this one also with some other models.

And when I tried to use the exported model, I received following error:
```
Fail: [ONNXRuntimeError] : 1 : FAIL : Load model from m2m100_onnx/decoder_model_merged.onnx failed:/onnxruntime_src/onnxruntime/core/graph/model.cc:146 onnxruntime::Model::Model(onnx::ModelProto&&, const PathString&, const IOnnxRuntimeOpSchemaRegistryList*, const onnxruntime::logging::Logger&, const onnxruntime::ModelOptions&) Unsupported model IR version: 9, max supported IR version: 8
```

I would be really thankful for some explanation or hints, since I can't find any information on how to fix this.

**Comment by claeyzre** [comment]
@goga334  I had the same issue.
The required onnx version of optimum are a bit loose w.r.t last onnx release. 
Try enforcing the following versions of Onnx when installing your dependencies:
```
onnxruntime = "1.13.1"
onnx = "1.13.1"
```

**Comment by regisss** [comment]
@goga334 Which version of Optimum do you use? With versions >= 1.8.4, this should be solved.

@claeyzre You should be able to use the latest version of ONNXRuntime (i.e. 1.14.1). And for the ONNX Python package, you're absolutely right, that's the fix we added in version 1.8.4 of Optimum.

**Comment by goga334** [comment]
@claeyzre @regisss, I've had `optimum==1.8.5` all the time. 
Interesting is that it works well with `onnxruntime==1.14.1` and `onnx==1.13.1`. 
However, with `onnxruntime==1.14.1` and `onnx==1.14.0` it crashes because of inconsistent IR versions, which was shown above.
